manager/state/raft/transport/peer.go

```diff
@@ -84,31 +111,44 @@ func (p *peer) send(m raftpb.Message) (err error) {
 	return nil
 }
 
-func (p *peer) update(addr string) error {
+func (p *peer) update(addrs ...string) error {
 	p.mu.Lock()
 	defer p.mu.Unlock()
-	if p.addr == addr {
-		return nil
+	var (
+		cc   *grpc.ClientConn
+		addr string
+	)
+
+	for _, a := range addrs {
+		c, err := p.tr.dial(a)
+		if err == nil {
+			addr = a
+			cc = c
+			break
+		}
+		log.G(p.ctx).WithError(err).Errorf("failed to create conn for %x with addr %s", p.id, a)
 	}
-	cc, err := p.tr.dial(addr)
-	if err != nil {
-		return err
+	if cc == nil {
+		return errors.New("dial failed for all provided addrs")
 	}
 
+	p.alternateAddrs = toAddrMap(addr, addrs)
+
 	p.cc.Close()
 	p.cc = cc
-	p.addr = addr
+	p.ccAddr = addr
 	return nil
 }
 
-func (p *peer) updateAddr(addr string) error {
+func (p *peer) updateAddr(addrs ...string) error {
 	p.mu.Lock()
 	defer p.mu.Unlock()
-	if p.addr == addr {
-		return nil
+	addrMap := make(map[string]struct{}, len(addrs))
+	for _, a := range addrs {
+		addrMap[a] = struct{}{}
 	}
-	log.G(p.ctx).Debugf("peer %x updated to address %s, it will be used if old failed", p.id, addr)
-	p.newAddr = addr
+	p.alternateAddrs = addrMap
+	log.G(p.ctx).Debugf("peer %x updated to addresses %v, it will be used if old failed", p.id, addrs)
```

Comment:
"peer %x updated with addresses %v"

Remove ", it will be used if old failed"

Did you confirm that the map renders legibly in the log output?